code/modules/xenoarcheaology/tools/tools.dm

```diff
@@ -310,3 +310,170 @@
 		usr << browse(null, "window=locater")
 
 	updateSelfDialog()
+
+/obj/item/device/xenoarch_multi_tool
```

Comment:
It might be preferable to store a device of each type _in_ the multitool, and then route all of the functions to _those_, instead of copypasta-ing all the code directly and adding a bit of spaghetti to switch between each function. Examples of what I mean below